### PR TITLE
Proof of Concept: Move 3 Unit Tests

### PR DESCRIPTION
**Pull request by ClaytonCalabrese**
This PR is the proof of concept of migrating the unit tests to their own directory as we defined in our documents.

### CODE REVIEW

File: tests/unit-tests/dds/DCPS/run_test_coverage.pl [hunk 1]
```diff
@@ -0,0 +1,79 @@
+eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
```
Comment (iguessthislldo) [inline tests/unit-tests/dds/DCPS/run_test_coverage.pl:1]: I'm not entirely sure what script file is for, but I don't think you needed to copy it.

File: tests/unit-tests/dds/security/AccessControlTest.cpp [hunk 2]
```diff
@@ -26,8 +26,8 @@ using namespace testing;
 static const char* Expected_Permissions_Token_Class_Id ="DDS:Access:Permissions:1.0";
```
Comment (iguessthislldo) [inline tests/unit-tests/dds/security/AccessControlTest.cpp:26]: Name should be `AccessControlBuiltInImpl.cpp` to match the code it's testing. The lint script will have to match the files using the names.

File: tests/unit-tests/dds/DCPS/run_test_unit.pl [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        my $executable;
+        if ($file =~ /$testExe/o) {
+          $executable = $1;
+          # each process runs to completion before the next starts
+          my $LONE_PROCESS = 1;
+          if ($executable eq "UnitTests_BIT_DataReader") {
+            $test->process("$executable", "$executable", "-DCPSConfigFile rtps.ini", $LONE_PROCESS);
+          } else {
+            $test->process("$executable", "$executable", "", $LONE_PROCESS);
+          }
```
Comment (iguessthislldo) [inline tests/unit-tests/dds/DCPS/run_test_unit.pl:59]:
```suggestion
          $test->process("$executable", "$executable", "", $LONE_PROCESS);
```

File: tests/unit-tests/dds/DCPS/run_test_unit.pl [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+      }
+    }
+    closedir($fh);
+  }
+}
+
+run_unit_tests();
+
+unless ($something_ran == 1) {
+  print STDERR "ERROR: no test was run\n";
```
Comment (iguessthislldo) [inline tests/unit-tests/dds/DCPS/run_test_unit.pl:80]:
```suggestion
  print STDERR "ERROR: no test was run\n";
  exit 1;
```

Not really relevant, but it's bothering me now that I saw it was saying there would be an error, but not returning a non-zero status.

File: tests/unit-tests/dds/DCPS/.gitignore [hunk 5]
```diff
@@ -0,0 +1,2 @@
+/UnitTests_DataSampleHeader
+/SafetyProfilePoolTest
```
Comment (iguessthislldo) [inline tests/unit-tests/dds/DCPS/.gitignore:2]: Not really necessary, but we should probably make the MPC project and executable names consistent when we merge everything together.

File: tests/dcps_tests.lst [hunk 6]
```diff
@@ -407,7 +408,7 @@ tests/DCPS/Instances/run_test.pl multiple_instance multiple_datawriter nokey: !D
 tests/DCPS/SkipSerialize/run_test.pl rtps_disc: !DCPS_MIN !NO_MCAST RTPS !DDS_NO_OWNERSHIP_PROFILE !OPENDDS_SAFETY_PROFILE
 tests/FACE/Unit/run_test.pl qossettings: !DCPS_MIN !WCHAR !NO_BUILT_IN_TOPICS
 tests/FACE/Unit/run_test.pl memorypool: !DCPS_MIN !WCHAR !NO_BUILT_IN_TOPICS
-tests/FACE/Unit/run_test.pl safetyprofilepool: !DCPS_MIN !WCHAR !NO_BUILT_IN_TOPICS OPENDDS_SAFETY_PROFILE
+tests/unit-tests/dds/DCPS/run_test_safety.pl safetyprofilepool: !DCPS_MIN !WCHAR !NO_BUILT_IN_TOPICS OPENDDS_SAFETY_PROFILE
```
Comment (iguessthislldo) [inline tests/dcps_tests.lst:411]: This removed the non-migrated FACE unit tests from the test lineup.

File: tests/unit-tests/dds/DCPS/.gitignore (hunk 5 above)
Comment (iguessthislldo) [inline tests/unit-tests/dds/DCPS/.gitignore:2]: Not really necessary at the moment, but we should probably make the MPC project and executable names consistent when we merge everything together.

File: tests/unit-tests/dds/security/AccessControlTest.cpp (hunk 2 above)
Comment (iguessthislldo) [inline tests/unit-tests/dds/security/AccessControlTest.cpp:26]: The line for this unit test is still the same in [`tests/security/security_tests.lst`](https://github.com/objectcomputing/OpenDDS/blob/7561732ba6d9edfa81485460962714c7ced29a4e/tests/security/security_tests.lst#L3).

**Review by iguessthislldo — CHANGES_REQUESTED** [review]

File: tests/dcps_tests.lst (hunk 6 above)
Comment (ClaytonCalabrese) [reply tests/dcps_tests.lst:411]: What do you mean by this? Are you saying the safetyprofilepool test should not be in the dcps test list now, or are you saying I should delete the other FACE tests from the list?

Comment (iguessthislldo) [reply tests/dcps_tests.lst:411]: You're right, I read that wrong. I thought it was running all the FACE unit tests in one entry.

File: tests/security/security_tests.lst [hunk 7]
```diff
@@ -1,6 +1,6 @@
 tests/security/IDL_Serialization/run_test.pl
 tests/security/SSL/run_test.pl
-tests/security/unit_tests/run_test.pl OpenDDS_AccessControlTest
+tests/unit-tests/dds/DCPS/run_test.pl AccessControlBuildInImpl
```
Comment (iguessthislldo) [inline tests/security/security_tests.lst:3]:
```suggestion
tests/unit-tests/dds/DCPS/security/run_test.pl AccessControlBuiltInImpl
```

File: tests/unit-tests/dds/DCPS/run_test_unit.pl [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+    my $testExe = "^(UnitTests_[^\.]*?)(:?\.exe)?\$";
+    if (opendir($fh, $dir)) {
+      foreach my $file (readdir($fh)) {
+        my $test = new PerlDDS::TestFramework();
+        my $executable;
+        if ($file =~ /$testExe/o) {
+          $executable = $1;
+          # each process runs to completion before the next starts
+          my $LONE_PROCESS = 1;
+         $test->process("$executable", "$executable", "", $LONE_PROCESS);
```
Comment (iguessthislldo) [inline tests/unit-tests/dds/DCPS/run_test_unit.pl:55]:
```suggestion
          $test->process("$executable", "$executable", "", $LONE_PROCESS);
```

**Review by iguessthislldo — CHANGES_REQUESTED** [review]

**Comment by iguessthislldo** [comment]
You committed this program `tests/unit-tests/dds/DCPS/security/OpenDDS_AccessControlTest` and there are issues with the test itself: https://github.com/objectcomputing/OpenDDS/pull/2456/files#annotation_1087502349.

**Comment by mitza-oci** [comment]
> 
> 
> You committed this program `tests/unit-tests/dds/DCPS/security/OpenDDS_AccessControlTest` and there are issues with the test itself: https://github.com/objectcomputing/OpenDDS/pull/2456/files#annotation_1087502349.

Please squash/rebase to get rid of the binary from history.

**Review by iguessthislldo — CHANGES_REQUESTED** [review]
I think it looks good enough for now. We still need to make the unit tests consistent with each other, but we can do that in the next phase. The only thing left to do is to remove the binary that was committed from the history.

**Review by iguessthislldo — APPROVED** [review]

**Pull request merged**